code/go/0chain.net/core/common/rate_limiter.go

```diff
@@ -1,159 +1,86 @@
 package common
 
 import (
+	"fmt"
 	"net/http"
 	"strings"
 	"time"
 
+	"github.com/0chain/blobber/code/go/0chain.net/core/logging"
 	tollbooth "github.com/didip/tollbooth/v6"
 	"github.com/didip/tollbooth/v6/limiter"
-	"github.com/spf13/viper"
-	rl "go.uber.org/ratelimit"
 )
 
-type ratelimit struct {
-	Limiter           *limiter.Limiter
-	RateLimit         bool
-	RequestsPerSecond float64
-}
-
-var userRateLimit *ratelimit
-var fileRateLimit *ratelimit
-
-func initUserRateLimiter(userRl float64) {
-	userRateLimit = &ratelimit{RequestsPerSecond: userRl}
+func GetRateLimiter(rps float64, ipLookups []string, ignoreUrl bool, tokExpireTTL time.Duration) *limiter.Limiter {
+	lmt := tollbooth.NewLimiter(rps, &limiter.ExpirableOptions{
+		DefaultExpirationTTL: tokExpireTTL,
+	})
 
-	if userRateLimit.RequestsPerSecond == 0 {
-		userRateLimit.RateLimit = false
-		return
+	if ipLookups != nil {
+		lmt.SetIPLookups(ipLookups)
 	}
 
-	userRateLimit.RateLimit = true
-	userRateLimit.Limiter = tollbooth.NewLimiter(userRateLimit.RequestsPerSecond, &limiter.ExpirableOptions{DefaultExpirationTTL: time.Hour}).
-		SetIPLookups([]string{"RemoteAddr", "X-Forwarded-For", "X-Real-IP"}).
-		SetMethods([]string{"GET", "POST", "PUT", "DELETE"})
-}
-
-func initFileRateLimiter(fileRl float64) {
-	fileRateLimit = &ratelimit{RequestsPerSecond: fileRl}
-
-	if fileRateLimit.RequestsPerSecond == 0 {
-		fileRateLimit.RateLimit = false
-		return
-	}
-
-	fileRateLimit.RateLimit = true
-	fileRateLimit.Limiter = tollbooth.NewLimiter(fileRateLimit.RequestsPerSecond, &limiter.ExpirableOptions{DefaultExpirationTTL: time.Hour}).
-		SetIPLookups([]string{}). // IP not required
-		SetIgnoreURL(false)       // URL Path contains allocation
-}
-
-//ConfigRateLimits - configure the rate limits
-func ConfigRateLimits() {
-	initUserRateLimiter(viper.GetFloat64("handlers.rate_limit"))
-	initFileRateLimiter(viper.GetFloat64("handlers.file_rate_limit"))
+	lmt.SetIgnoreURL(ignoreUrl)
+	return lmt
 }
 
-func NewGRPCRateLimiter() *GRPCRateLimiter {
-	userRl := viper.GetFloat64("handlers.rate_limit")
+func RateLimit(handler ReqRespHandlerf, lmt *limiter.Limiter) ReqRespHandlerf {
+	return func(w http.ResponseWriter, r *http.Request) {
 
-	return &GRPCRateLimiter{rl.New(int(userRl))}
-}
+		if lmt != nil {
 
-type GRPCRateLimiter struct {
-	rl.Limiter
-}
+			keys := tollbooth.BuildKeys(lmt, r)
+			clientID := r.Header.Get(ClientHeader)
 
-func (r *GRPCRateLimiter) Limit() bool {
-	r.Take()
-	return false
-}
-
-//UserRateLimit - rate limiting for end user handlers
-func UserRateLimit(handler ReqRespHandlerf) ReqRespHandlerf {
-	if !userRateLimit.RateLimit {
-		return handler
-	}
-	return func(writer http.ResponseWriter, request *http.Request) {
-		tollbooth.LimitFuncHandler(userRateLimit.Limiter, handler).ServeHTTP(writer, request)
-	}
-}
+			keys = append(keys, []string{clientID})
```

Comment:
Check if clientID is empty before append it to keys.

Reply:
I think we should also have empty client key.
We must mandate presence of clientID in the header.
Only some endpoints don't have clientID requirement. But we should verify clientID as well later on. So empty clientID as a key in rate-limiter will prevent un-necessary access.